vars/withNodeJSEnvUnix.groovy

```diff
@@ -28,35 +28,82 @@ def call(Map args = [:], Closure body) {
     "HOME=${env.WORKSPACE}"
   ]){
     def node_version = installNode(args)
-    withEnv(["PATH+NVM=${HOME}/.nvm/versions/node/${node_version}/bin"]){
+    def nvmNodePath = getNodePath(node_version)
+    if (!fileExists(nvmNodePath)) {
+      error('withNodeJSEnvUnix: the node version could not be found! Look for errors in the logs.')
+    }
+    withEnv(["PATH+NVM=${nvmNodePath}"]){
+      sh(label: 'Debug withNodeJSEnvUnix installation', script: """
+        which npm || true
+        whereis npm || true
+        npm --version || true
+        which node || true
+        whereis node || true
+        node --version || true
+        ls -l ${nvmNodePath} || true
+      """)
       body()
     }
   }
 }
 
 def installNode(Map args = [:]) {
   def version = args.containsKey('version') ? args.version : nodeJSDefaultVersion()
-  def nodeVersionLocation = pwd(tmp: true)
+  def nvmVersion = args.get('nvmVersion', 'v0.39.2')
+  def nvmNodeFile = nvmNodeVersionFile()
   retryWithSleep(retries: 3, seconds: 5, backoff: true){
-    sh(label: 'Installing nvm', script: '''
+    sh(label: 'Installing nvm', script: """
       set -e
-      export NVM_DIR="${HOME}/.nvm"
-      [ -s "${NVM_DIR}/nvm.sh" ] && . "${NVM_DIR}/nvm.sh"
+      set +x
+      export NVM_DIR="\${HOME}/.nvm"
+      if [ -s "\${NVM_DIR}/nvm.sh" ] ; then
+        echo "load existing nvm environment"
+        . "\${NVM_DIR}/nvm.sh"
+      fi
 
-      if [ -z "$(command -v nvm)" ]; then
-        rm -fr "${NVM_DIR}"
-        curl -so- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
+      if [ -z "\$(command -v nvm)" ]; then
+        echo "installing nvm"
+        rm -fr "\${NVM_DIR}"
+        curl -so- https://raw.githubusercontent.com/nvm-sh/nvm/${nvmVersion}/install.sh | bash
       fi
-    ''')
+    """)
     sh(label: "Installing Node.js ${version}", script: """
       set -e
+      set +x
+      echo "load existing nvm environment"
       export NVM_DIR="\${HOME}/.nvm"
       [ -s "\${NVM_DIR}/nvm.sh" ] && . "\${NVM_DIR}/nvm.sh"
 
-      nvm install ${version}
-      nvm version | head -n1 > "${nodeVersionLocation}/.nvm-node-version"
+      echo "install node.js"
+      nvm install --no-progress --default ${version}
+      nvm use ${version}
+
+      echo "Debug nvm env"
+      nvm ls
+      nvm version
+      nvm --version
+      nvm current
+      nvm version | head -n1
+
+      echo "Fetch the default nodejs version installed with nvm"
+      nvm version | head -n1 > "${nvmNodeFile}"
     """)
   }
+  def nvmNodeVersion = readFile(file: nvmNodeFile).trim()
+
+  // In some cases, nvm cannot resolve the installed version,
+  // let's then fallback to a hardcoded value.
+  if (!nvmNodeVersion.trim()) {
```

Comment:
For some reason  when installing `18.x` nodejs with `nvm` the command `nvm version` does not report any value while it works completely fine with version >= `17.x`

This should help with falling back to the given version with the nvm format.